File: 1-100/38.count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        ret = '1'
        for _ in range(n-1):
            tmp = ''
            cur = ret[0]
            nCnt = 1
            for c in ret[1:]:
                if cur == c:
                    nCnt += 1
                else:
                    tmp += str(nCnt) + cur
                    nCnt = 1
                    cur = c
            tmp += str(nCnt) + cur

            ret = tmp
        
        return ret

File: 1-100/38.count-and-say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string ret = "1";
        while (--n > 0)
        {
            string tmp = "";
            int nCnt = 1;
            char cur = ret[0];
            size_t nSize = ret.size();
            for (size_t i = 1; i < nSize; ++i)
            {
                if (cur == ret[i])
                {
                    ++nCnt;
                }
                else
                {
                    tmp += nCnt + '0';
                    nCnt = 1;
                    tmp += cur;
                    cur = ret[i];
                }
            }
            tmp += nCnt + '0';
            tmp += cur;

            ret = tmp;
        }

        return ret;
    }
};